### PR TITLE
feat(global): Isolate extensions and make it compatible with cumcumber 3

### PR DESCRIPTION
**Pull request by bnadim**
**Summary**
Isolated extensions context per world rather than shared module

**Test plan**

### CODE REVIEW

File: src/extensions/fixtures/fixtures_loader.js [hunk 1]
```diff
@@ -11,183 +11,214 @@ const chalk = require('chalk')
...
  *
- * @param {string} _fixturesDir - The name of the fixtures directory relative to feature
+ * @class
  */
-exports.configure = ({ fixturesDir: _fixturesDir = 'fixtures' } = {}) => {
-    fixturesDir = _fixturesDir
-}
+class FixturesLoader {
+    /**
+     * @param {string} _fixturesDir - The name of the fixtures directory relative to feature
```
Comment (plouc) [inline src/extensions/fixtures/fixtures_loader.js]: `@param {string} [_fixturesDir='fixtures'] - The name of the fixtures directory relative to feature`

File: src/extensions/fixtures/fixtures_loader.js [hunk 2]
```diff
@@ -11,183 +11,214 @@ const chalk = require('chalk')
...
- *
- * @param {string} _featureUri - Feature uri
- */
-exports.setFeatureUri = _featureUri => {
-    featureUri = _featureUri
-}
+    /**
+     * Configures the loader
+     *
+     * @param {string} _fixturesDir - The name of the fixtures directory relative to feature
```
Comment (plouc) [inline src/extensions/fixtures/fixtures_loader.js]: same here

Comment (plouc) [reply src/extensions/fixtures/fixtures_loader.js]: You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

File: src/extensions/cli/cli.js [hunk 3]
```diff
@@ -10,182 +10,211 @@ const { spawn } = require('child_process')
...
- *
- * @type {Object}
- */
-let env = {}
+    /**
+     * Sets the Current Working Directory for the command.
+     *
+     * @param {string} _cwd - The new CWD
+     */
+    setCwd(_cwd) {
```
Comment (plouc) [inline src/extensions/cli/cli.js]: You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

File: src/extensions/cli/cli.js [hunk 4]
```diff
@@ -10,182 +10,211 @@ const { spawn } = require('child_process')
...
 
-/** @type {number} */
-let killDelay
+    /**
+     * Defines environment variables.
+     * Beware that all existing ones will be overridden!
+     *
+     * @param {Object} _env - The environment variables object
+     */
+    setEnvironmentVariables(_env) {
```
Comment (plouc) [inline src/extensions/cli/cli.js]: You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

File: src/extensions/fixtures/fixtures_loader.js [hunk 5]
```diff
@@ -11,183 +11,214 @@ const chalk = require('chalk')
...
+    /**
+     * Sets feature uri, used to resolve fixtures files.
+     * When trying to load a fixture file the path will be comprised of:
+     * - feature uri
+     * - fixturesDir
+     * - fixture name
+     *
+     * @param {string} _featureUri - Feature uri
+     */
+    setFeatureUri(_featureUri) {
```
Comment (plouc) [inline src/extensions/fixtures/fixtures_loader.js]: You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

File: src/extensions/http_api/client.js [hunk 6]
```diff
@@ -13,230 +13,256 @@ const { Cookie } = require('tough-cookie')
...
- * Enables cookie jar.
- */
-exports.enableCookies = () => {
-    if (cookieJar !== null) return
+    /**
+     * Sets request query parameters.
+     *
+     * @param {Object} _query
+     */
+    setQuery(_query) {
```
Comment (plouc) [inline src/extensions/http_api/client.js]: You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

File: src/extensions/http_api/client.js [hunk 7]
```diff
@@ -13,230 +13,256 @@ const { Cookie } = require('tough-cookie')
...
 
-    cookieJar = request.jar()
-    cookieJar._jar.rejectPublicSuffixes = false
-}
+    /**
+     * Sets request headers.
+     *
+     * @param {Object} _headers
+     */
+    setHeaders(_headers) {
```
Comment (plouc) [inline src/extensions/http_api/client.js]: You can remove the `_`, it was only there to not conflict with var name which is now bound to `this`

**Comment by plouc** [comment]
@bnadim, thank you! just a few micro adjustments.

File: src/extensions/fixtures/fixtures_loader.js (hunk 1 above)
Comment (bnadim) [reply src/extensions/fixtures/fixtures_loader.js]: Ok

File: src/extensions/fixtures/fixtures_loader.js (hunk 2 above)
Comment (bnadim) [reply src/extensions/fixtures/fixtures_loader.js]: ok

File: src/extensions/fixtures/fixtures_loader.js (hunk 5 above)
Comment (bnadim) [reply src/extensions/fixtures/fixtures_loader.js]: ok

File: src/extensions/cli/cli.js (hunk 3 above)
Comment (bnadim) [reply src/extensions/cli/cli.js]: ok

File: src/extensions/http_api/client.js (hunk 6 above)
Comment (bnadim) [reply src/extensions/http_api/client.js]: ok

File: src/extensions/http_api/client.js (hunk 7 above)
Comment (bnadim) [reply src/extensions/http_api/client.js]: ok

**Comment by bnadim** [comment]
If you can review please, i followed the reviews

**Comment by bnadim** [comment]
@plouc i fixed what you asked, fixed compatibility issue with cumcumber 3 and update travis to support cucumber 2 and 3.
If it's ok for you please merge it when you can :)

**Pull request merged**